losses/base.py

```diff
@@ -184,10 +186,33 @@ def __call__(self, x, y):
         return x, y.to(self.device)
 
 
+class ModelLoader:
+
+    def __init__(self, loss_name):
+        self.loss_name = loss_name
+
+    def restore(self, model: SimNet, loss_fn: nn.Module, optimizer: Optimizer, path: str):
+        checkpoint = torch.load(path)
+        model.load_common_state_dict(checkpoint['common_state_dict'])
+        if self.loss_name == checkpoint['trained_loss']:
+            loss_fn.load_state_dict(checkpoint['loss_state_dict'])
+            if model.loss_module is not None:
+                model.loss_module.load_state_dict(checkpoint['loss_module_state_dict'])
+            optimizer.load_state_dict(checkpoint['optim_state_dict'])
+        print(f"Recovered Model. Epoch {checkpoint['epoch']}. Test Metric {checkpoint['accuracy']}")
+        return checkpoint
+
+    def load(self, model: SimNet, path: str):
+        checkpoint = torch.load(path)
+        model.load_common_state_dict(checkpoint['common_state_dict'])
+        if self.loss_name == checkpoint['trained_loss'] and model.loss_module is not None:
```

Comment:
Since this method is only used for loading prediction models, maybe we can forget about the loss module.
All evaluation methods will use the final embeddings instead of the loss module predictions